src/Streamly/Internal/Data/Array/Foreign/Mut/Types.hs

```diff
@@ -524,6 +526,16 @@ read = Unfold step inject
             return $ D.Yield x
                 (ReadUState fp (p `plusPtr` sizeOf (undefined :: a)))
 
+    extract (ReadUState (ForeignPtr end contents) (Ptr p)) =
+        return $ Array (ForeignPtr p contents) (Ptr end) (Ptr end)
```

Comment:
The type of `extact :: s -> m a` puts a constraint on a few things. I might want to return the entire array along with the index (ptr) till where we generated. No use-case but...

I'm sure we'll encounter a use case that forces us to change the type of `extract` to `s -> m c`.